src/main/java/org/socialhistoryservices/delivery/reservation/controller/ReservationController.java

```diff
@@ -234,6 +255,28 @@ private String processVisitorReservationCreation(HttpServletRequest req, Reserva
         return "reservation_create";
     }
 
+    private ArrayList<String> getAllAvailableHoldingsInReservation(Reservation newRes) {
+        List<HoldingReservation> holdingReservations = newRes.getHoldingReservations();
+        ArrayList<String> listOfPids = new ArrayList<>();
+        for (HoldingReservation hr : holdingReservations) {
+            try {
+                ExternalRecordInfo eri = lookup.getRecordMetaDataByPid(hr.getHolding().getRecord().getPid());
```

Comment:
Don't use the api on this level. Everything should be in the database, use that information. You already have a method written for this purpose: `getListOfSiblingHoldingInSameContainer`